src/localisation/localisation.c

```diff
@@ -1316,6 +1317,20 @@ money32 string_to_money(char * string_to_monetise)
 		}
 	}
 	
+	//if first character of shortened string is a minus, consider number negative
+	if (text_ptr[0] == '-') {
```

Comment:
It looks like you never initialise `sign` when the number is positive.
It would suffice to add an `else` clause to this if-block, setting `sign` to `1`.